src/main/java/org/example/homework03/validator/UserValidator.java

```diff
@@ -0,0 +1,100 @@
+package org.example.homework03.validator;
+
+import org.example.homework03.domain.service.UserService;
+import org.example.homework03.dto.UserRegistrationDto;
+import org.example.homework03.dto.UserResponseDto;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+public class UserValidator {
+
+    private String error;
```

Comment:
What if several errors could be present during validation?
```suggestion
    private List<String> errors;
```